### PR TITLE
Replace "option" in doc when it refers to a flag

### PR DESCRIPTION
**Pull request by jfehrle**
Make wording in the doc more consistent.

Flags are boolean, options are non-boolean and their syntax is different.

### CODE REVIEW

File: doc/sphinx/addendum/type-classes.rst [hunk 1]
```diff
@@ -560,8 +560,8 @@ Settings
 
    Determines how much information is shown for typeclass resolution steps during search.
    1 is the default level.  2 shows additional information such as tried tactics and shelving
-   of goals.  Setting this option to 1 or 2 turns on :flag:`Typeclasses Debug`; setting this
-   option to 0 turns that option off.
+   of goals.  Setting this option to 1 or 2 turns on flag :flag:`Typeclasses Debug`; setting this
```
Comment (Zimmi48) [inline doc/sphinx/addendum/type-classes.rst]: Wording suggestion for consistency with other parts of the manual referencing flags:

```suggestion
   of goals.  Setting this option to 1 or 2 turns on the :flag:`Typeclasses Debug` flag; setting this
```

File: doc/sphinx/addendum/universe-polymorphism.rst [hunk 2]
```diff
@@ -480,7 +480,7 @@ underscore or by omitting the annotation to a polymorphic definition.
       About foo.
 
    To recover the same behaviour with regard to universes as
-   :g:`Defined`, the option :flag:`Private Polymorphic Universes` may
+   :g:`Defined`, the flag :flag:`Private Polymorphic Universes` may
```
Comment (Zimmi48) [inline doc/sphinx/addendum/universe-polymorphism.rst]:
```suggestion
   :g:`Defined`, the :flag:`Private Polymorphic Universes` flag may
```

File: doc/sphinx/language/cic.rst [hunk 3]
```diff
@@ -1224,7 +1224,7 @@ Conversion is preserved as any (partial) instance :math:`I_j~q_1 … q_r` or
 
 .. flag:: Template Check
 
-   Unsetting option :flag:`Template Check` disables the check of
+   Unsetting flag :flag:`Template Check` disables the check of
```
Comment (Zimmi48) [inline doc/sphinx/language/cic.rst]:
```suggestion
   This flag is on by default. Turning it off disables the check of
```

File: doc/sphinx/language/gallina-extensions.rst [hunk 4]
```diff
@@ -2299,7 +2299,7 @@ Printing universes
    This command can be used to print the constraints on the internal level
    of the occurrences of :math:`\Type` (see :ref:`Sorts`).
 
-   If the optional ``Sorted`` option is given, each universe will be made
+   If the ``Sorted`` option is given, each universe will be made
```
Comment (Zimmi48) [inline doc/sphinx/language/gallina-extensions.rst]: What about:\

```suggestion
   If the ``Sorted`` keyword is present, each universe will be made
```

Maybe not a keyword from a parsing perspective though. But option is not the right term here, since we have given it a precise meaning in this manual.

File: doc/sphinx/language/gallina-specification-language.rst [hunk 5]
```diff
@@ -779,7 +779,7 @@ Simple inductive types
       The types of the constructors have to satisfy a *positivity condition*
       (see Section :ref:`positivity`). This condition ensures the soundness of
       the inductive definition. The positivity checking can be disabled using
-      the option :flag:`Positivity Checking` (see :ref:`controlling-typing-flags`).
+      the flag :flag:`Positivity Checking` (see :ref:`controlling-typing-flags`).
```
Comment (Zimmi48) [inline doc/sphinx/language/gallina-specification-language.rst]:
```suggestion
      the :flag:`Positivity Checking` flag (see :ref:`controlling-typing-flags`).
```

File: doc/sphinx/language/gallina-specification-language.rst [hunk 6]
```diff
@@ -991,7 +991,7 @@ Variants
    The :cmd:`Variant` command is identical to the :cmd:`Inductive` command, except
    that it disallows recursive definition of types (for instance, lists cannot
    be defined using :cmd:`Variant`). No induction scheme is generated for
-   this variant, unless option :flag:`Nonrecursive Elimination Schemes` is on.
+   this variant, unless flag :flag:`Nonrecursive Elimination Schemes` is on.
```
Comment (Zimmi48) [inline doc/sphinx/language/gallina-specification-language.rst]:
```suggestion
   this variant, unless the :flag:`Nonrecursive Elimination Schemes` flag is on.
```

File: doc/sphinx/language/gallina-specification-language.rst [hunk 7]
```diff
@@ -1393,11 +1393,11 @@ Chapter :ref:`Tactics`. The basic assertion command is:
       The name you provided is already defined. You have then to choose
       another name.
 
-   .. exn:: Nested proofs are not allowed unless you turn option Nested Proofs Allowed on.
+   .. exn:: Nested proofs are not allowed unless you turn flag :flag:`Nested Proofs Allowed` on.
```
Comment (Zimmi48) [inline doc/sphinx/language/gallina-specification-language.rst]: You should not change this line unless you fix the error message directly in the code.

File: doc/sphinx/language/gallina-specification-language.rst [hunk 8]
```diff
@@ -1393,11 +1393,11 @@ Chapter :ref:`Tactics`. The basic assertion command is:
...
 
       You are asserting a new statement while already being in proof editing mode.
       This feature, called nested proofs, is disabled by default.
-      To activate it, turn option :flag:`Nested Proofs Allowed` on.
+      To activate it, turn flag :flag:`Nested Proofs Allowed` on.
```
Comment (Zimmi48) [inline doc/sphinx/language/gallina-specification-language.rst]:
```suggestion
      To activate it, turn the :flag:`Nested Proofs Allowed` flag on.
```

File: doc/sphinx/language/gallina-specification-language.rst [hunk 9]
```diff
@@ -1470,7 +1470,7 @@ using the keyword :cmd:`Qed`.
 
 .. note::
 
-   #. Several statements can be simultaneously asserted provided option
+   #. Several statements can be simultaneously asserted provided flag
```
Comment (Zimmi48) [inline doc/sphinx/language/gallina-specification-language.rst]: Can't use a one-line suggestion here.

File: doc/sphinx/proof-engine/ltac.rst [hunk 10]
```diff
@@ -860,7 +860,7 @@ We can carry out pattern matching on terms with:
       If the evaluation of the right-hand-side of a valid match fails, the next
       matching subterm is tried. If no further subterm matches, the next clause
       is tried. Matching subterms are considered top-bottom and from left to
-      right (with respect to the raw printing obtained by setting option
+      right (with respect to the raw printing obtained by setting flag
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/ltac.rst]: Ditto.

File: doc/sphinx/proof-engine/tactics.rst [hunk 11]
```diff
@@ -477,7 +477,7 @@ that occurrences have to be selected in the hypotheses named :token:`ident`.
 If no numbers are given for hypothesis :token:`ident`, then all the
 occurrences of :token:`term` in the hypothesis are selected. If numbers are
 given, they refer to occurrences of :token:`term` when the term is printed
-using option :flag:`Printing All`, counting from left to right. In particular,
+using flag :flag:`Printing All`, counting from left to right. In particular,
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/tactics.rst]:
```suggestion
using the :flag:`Printing All` flag, counting from left to right. In particular,
```

File: doc/sphinx/proof-engine/tactics.rst [hunk 12]
```diff
@@ -1527,7 +1527,7 @@ name of the variable (here :g:`n`) is chosen based on :g:`T`.
 
    This is equivalent to :n:`generalize @term` but it generalizes only over the
    specified occurrences of :n:`@term` (counting from left to right on the
-   expression printed using option :flag:`Printing All`).
+   expression printed using flag :flag:`Printing All`).
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/tactics.rst]:
```suggestion
   expression printed using the :flag:`Printing All` flag).
```

File: doc/sphinx/user-extensions/proof-schemes.rst [hunk 13]
```diff
@@ -128,7 +128,7 @@ Automatic declaration of schemes
 
 .. warning::
 
-   You have to be careful with this option since Coq may now reject well-defined
+   You have to be careful with this flag since Coq may now reject well-defined
```
Comment (Zimmi48) [inline doc/sphinx/user-extensions/proof-schemes.rst]:
```suggestion
   You have to be careful with these flags since Coq may now reject well-defined
```

Would be better to indent the `.. warning::` block within the `.. flag::` block to better indicate the scope of the warning.

**Review by Zimmi48 — APPROVED** [review]
That's what we had agreed upon. Thanks for taking care of this!

Feel free to ignore most of my rewording suggestions. Only one point is important, about changing the documentation of an existing error message.

**Comment by SkySkimmer** [comment]
TBH I liked calling them options better

**Comment by jfehrle** [comment]
Updated.  I took your suggestions, and added a handful of other changes, both in separate commits.  I'll squash after you take another look.

File: doc/sphinx/language/gallina-extensions.rst (hunk 4 above)
Comment (jfehrle) [reply doc/sphinx/language/gallina-extensions.rst]: "option" is also used to refer to command-line options.  Some overloading of terms is inevitable.

Comment (Zimmi48) [reply doc/sphinx/language/gallina-extensions.rst]: Yes, but in this case, I try to insist on using the complete phrasing "command-line option" or "command-line flag" for disambiguation.

**Comment by Zimmi48** [comment]
> TBH I liked calling them options better

Initially me too, but disambiguating between different behaviors seemed important enough to reinforce the specific "flag" terminology (that has been present, but not very much used, since forever). This PR can be seen as a follow up of #8439.

File: doc/sphinx/language/gallina-specification-language.rst [hunk 14]
```diff
@@ -1393,11 +1393,11 @@ Chapter :ref:`Tactics`. The basic assertion command is:
       The name you provided is already defined. You have then to choose
       another name.
 
-   .. exn:: Nested proofs are not allowed unless you turn option Nested Proofs Allowed on.
+   .. exn:: Nested proofs are not allowed unless you turn the :flag:`Nested Proofs Allowed` flag on.
```
Comment (Zimmi48) [inline doc/sphinx/language/gallina-specification-language.rst]: Sorry to insist: please undo the change on this line, or fix the error message in the code as well. Documentation of error messages must reflect exactly their content, even if the contents are flawed.

**Review by Zimmi48 — DISMISSED** [review]
~~Only a very small point remaining. Feel free to fix and squash in one go so that I can merge right after.~~

**Review by Zimmi48 — APPROVED** [review]
Looks good! Will merge as soon as this is squashed.

**Comment by jfehrle** [comment]
> I'm just stupid apparently.

To err is human, no need to beat yourself up.

**Comment by jfehrle** [comment]
Squashed.
